salt/_modules/metalk8s_network.py

```diff
@@ -0,0 +1,45 @@
+'''Metalk8s network related utilities.'''
+
+import ipaddress
+import logging
+
+from salt.exceptions import CommandExecutionError
+
+log = logging.getLogger(__name__)
+
+__virtualname__ = 'metalk8s_network'
+
+
+def __virtual__():
+    return __virtualname__
+
+
+def pick_nth_ip(service_network, index):
+
+    '''
+    Pick the coredns and kubernetes cluster ip address from service network.
+
+    :param service_network (str): CIDR presentation format read from pillars.
+     For example, '192.168.1.0/24'.
+    :param index (int): Kubernetes cluster IP address is set to 2nd address in
+     service_network.
+     coredns cluster IP address is set to the 10th address in the
+     service network
+
+    :returns (str):
+        An index param of 10 returns => coredns cluster ip
```

Comment:
I think we had clearly defined what an index `param` signifies.

Also FYI:
```
 :param index (int): Kubernetes cluster IP address is set to 2nd address in
     service_network.
     coredns cluster IP address is set to the 10th address in the
     service network
```
As such index=2 => we are computing the ip address for cluster DNS

I do not think it would be any better creating two functions that perform the same actions and so this is the reason we have one generic function doing the same action.

Would mark this as resolved unless another thinks otherwise

Reply:
Maybe it should read `An index value of 10 returns = CoreDNS cluster IP`